RFS/src/main/java/org/opensearch/migrations/bulkload/version_os_2_11/IndexCreator_OS_2_11.java

```diff
@@ -40,35 +43,45 @@ public boolean create(
         body.set("mappings", index.getMappings());
         body.set("settings", settings);
 
-        // Create the index; it's fine if it already exists
         try {
-            if (mode == MigrationMode.SIMULATE) {
-                return !client.hasIndex(index.getName());
-            } else if (mode == MigrationMode.PERFORM) {
-                return client.createIndex(index.getName(), body, context).isPresent();
-            }
-        } catch (InvalidResponse invalidResponse) {
-            var illegalArguments = invalidResponse.getIllegalArguments();
+            // Create the index; it's fine if it already exists
+            try {
+                var alreadyExists = false;
+                if (mode == MigrationMode.SIMULATE) {
+                    alreadyExists = client.hasIndex(index.getName());
+                } else if (mode == MigrationMode.PERFORM) {
+                    alreadyExists = client.createIndex(index.getName(), body, context).isEmpty();
+                }
 
-            if (illegalArguments.isEmpty()) {
-                log.debug("Cannot retry invalid response, there are no illegal arguments to remove.");
-                throw invalidResponse;
-            }
+                if (alreadyExists) {
+                    result.failureType(CreationFailureType.ALREADY_EXISTS);
+                }
+            } catch (InvalidResponse invalidResponse) {
+                var illegalArguments = invalidResponse.getIllegalArguments();
 
-            for (var illegalArgument : illegalArguments) {
-                if (!illegalArgument.startsWith("index.")) {
-                    log.warn("Expecting all retryable errors to start with 'index.', instead saw " + illegalArgument);
+                if (illegalArguments.isEmpty()) {
+                    log.debug("Cannot retry invalid response, there are no illegal arguments to remove.");
                     throw invalidResponse;
                 }
 
-                var shortenedIllegalArgument = illegalArgument.replaceFirst("index.", "");
-                removeFieldsByPath(settings, shortenedIllegalArgument);
-            }
+                for (var illegalArgument : illegalArguments) {
+                    if (!illegalArgument.startsWith("index.")) {
+                        log.warn("Expecting all retryable errors to start with 'index.', instead saw " + illegalArgument);
+                        throw invalidResponse;
```

Comment:
Should we instead include an exception with  the message above, otherwise this log may be lost in the output